src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BlockListener.java

```diff
@@ -237,24 +238,30 @@ private void callBlockHandler(BlockBreakEvent e, ItemStack item, List<ItemStack>
     }
 
     @ParametersAreNonnullByDefault
-    private void dropItems(BlockBreakEvent e, List<ItemStack> drops) {
+    private void dropItems(BlockBreakEvent e, ItemStack item, Block block, @Nullable SlimefunItem sfBlock, List<ItemStack> drops) {
         if (!drops.isEmpty()) {
             // TODO: properly support loading inventories within unit tests
             if (!Slimefun.instance().isUnitTest()) {
                 // Notify plugins like CoreProtect
-                Slimefun.getProtectionManager().logAction(e.getPlayer(), e.getBlock(), Interaction.BREAK_BLOCK);
+                Slimefun.getProtectionManager().logAction(e.getPlayer(), block, Interaction.BREAK_BLOCK);
             }
 
             // Fixes #2560
             if (e.isDropItems()) {
                 // Disable normal block drops
                 e.setDropItems(false);
 
+                // Fixes #4051
+                if (sfBlock == null) {
+                    block.breakNaturally(item);
```

Comment:
call Block#breakNaturally will drop items, it will caused many bugs(double drops,drops that should disappear fall normally)